python-ecosys/urequests/urequests.py

```diff
@@ -55,11 +63,16 @@ def request(method, url, data=None, json=None, headers={}, stream=None):
     ai = usocket.getaddrinfo(host, port, 0, usocket.SOCK_STREAM)
     ai = ai[0]
 
+    resp_d = None
+    if parse_headers is not False:
+        resp_d = {}
+
     s = usocket.socket(ai[0], ai[1], ai[2])
```

Comment:
Is the socked always closed correctly? Should we use a context manager to control the lifetime of the socket?

Reply:
I'm not sure how this should be handled to be honest. It needs to be left open at the return of the `request()` function as the caller will generally read the content later.

The `Response()` object has the close function, it probably should be updated with __enter__ and __exit__ so it can be used as a context manager the same as in cpython requests.

Reply:
Ref: https://github.com/micropython/micropython-lib/pull/278